src/quoter.ts

```diff
@@ -42,34 +64,35 @@ export abstract class SwapQuoter {
     const singleHop = route.pools.length === 1
     const quoteAmount: string = toHex(amount.quotient)
     let calldata: string
+    const swapInterface = options.useQuoterV2 ? new Interface(IQuoterV2.abi) : this.INTERFACE
```

Comment:
this should be statically accessed and only constructed once. if you want to only do it when necessary, could write a custom get to cache the construction